packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -101,16 +121,53 @@ export default function MongoNav({
   admin = false,
   constructOrganizationURL: constructOrganizationURLProp,
   constructProjectURL: constructProjectURLProp,
+  onError = () => {},
+  onSuccess = () => {},
 }: MongoNavInterface) {
-  let data;
+  const [data, setData] = React.useState<DataInterface | undefined>(undefined);
 
-  if (mode === Mode.Dev) {
-    data = devModeData;
+  const accountHost = hosts?.account ?? `https://account.mongodb.com`;
+  const cloudHost = hosts?.cloud ?? `https://cloud.mongodb.com`;
+  const universityHost = hosts?.university ?? `https://university.mongodb.com`;
+  const supportHost = hosts?.support ?? `https://support.mongodb.com`;
+
+  const endpointUri = `${cloudHost}/user/shared`;
+  const getProductionData = () =>
+    fetch(endpointUri, {
+      credentials: 'include',
+      method: 'GET',
+    });
+
+  function getFixtureData() {
+    return new Promise(resolve => {
+      setTimeout(() => {
+        onSuccess?.(fixtureData);
+
+        resolve(fixtureData);
+      }, 1000);
+    });
   }
 
-  if (!data) {
-    // Eventually this logic will be more robust, but for an alpha version will return null without data
-    return null;
+  useEffect(() => {
+    if (mode === Mode.Dev) {
+      getFixtureData().then(data => setData(data as DataInterface));
+    } else {
+      getProductionData()
+        .then(response => response.json())
+        .then(data => {
+          setData(data);
+          onSuccess?.(data);
+        })
+        .catch(error => {
+          onError?.(error);
+          console.error(error);
+        });
+    }
+  }, []);
```

Comment:
Might want to have this re-run if `mode` or `endpointUri` changes.

Reply:
👍